src/main/java/org/aksw/iguana/cc/worker/impl/SPARQLProtocolWorker.java

```diff
@@ -0,0 +1,412 @@
+            if (completed() && exception.isEmpty())
+                return (response.get().statusCode() / 100) == 2;
+            return false;
+        }
+    }
+
+
+    private HttpClient httpClient;
+    private final ThreadPoolExecutor executor;
+
+    private final XXHashFactory hasherFactory = XXHashFactory.fastestJavaInstance();
+    private final RequestFactory requestFactory;
+
+    private final ResponseBodyProcessor responseBodyProcessor;
+
+    // declared here, so it can be reused across multiple method calls
+    private BigByteArrayOutputStream responseBodybbaos = new BigByteArrayOutputStream();
+
+    // used to read the http response body
+    private final byte[] buffer = new byte[4096];
+
+    private final static Logger LOGGER = LoggerFactory.getLogger(SPARQLProtocolWorker.class);
+
+    @Override
+    public Config config() {
+        return (SPARQLProtocolWorker.Config) config;
+    }
+
+
+    public SPARQLProtocolWorker(long workerId, ResponseBodyProcessor responseBodyProcessor, Config config) {
+        super(workerId, responseBodyProcessor, config);
+        this.responseBodyProcessor = responseBodyProcessor;
+        this.executor = (ThreadPoolExecutor) Executors.newFixedThreadPool(1);
+        this.requestFactory = new RequestFactory(config().requestType());
+        this.httpClient = buildHttpClient();
+    }
+
+
+    public CompletableFuture<Result> start() {
+        return CompletableFuture.supplyAsync(() -> {
+            List<ExecutionStats> executionStats = new ArrayList<>();
+            if (config().completionTarget() instanceof QueryMixes queryMixes) {
+                for (int i = 0; i < queryMixes.number(); i++) {
+                    for (int j = 0; j < config().queries().getQueryCount(); j++) {
+                        ExecutionStats execution = executeQuery(config().timeout(), false);
+                        if (execution != null) {
+                            logExecution(execution);
+                            executionStats.add(execution);
+                        }
+                    }
+                    LOGGER.info("{}\t:: Completed {} out of {} querymixes", this, i + 1, queryMixes.number());
+                }
+            } else if (config().completionTarget() instanceof TimeLimit timeLimit) {
+                final Instant endTime = Instant.now().plus(timeLimit.duration());
+                Instant now;
+                while ((now = Instant.now()).isBefore(endTime)) {
+                    final Duration timeToEnd = Duration.between(now, endTime);
+                    final boolean timeoutBeforeEnd = config().timeout().compareTo(timeToEnd) < 0;
+                    final Duration thisQueryTimeOut = (timeoutBeforeEnd) ? config().timeout() : timeToEnd;
+                    // If timeoutBeforeEnd is false, fail shouldn't be counted as timeout
+                    ExecutionStats execution = executeQuery(thisQueryTimeOut, !timeoutBeforeEnd);
+                    if (execution != null){
```

Comment:
comment for if

Reply:
I don't know exactly what you meant with this comment, but I'm just going to assume, that you meant that I should move the comment to the if statement. 😅